java/clients/src/main/java/sleeper/status/report/job/query/JobQuery.java

```diff
@@ -37,6 +37,9 @@ static JobQuery from(String tableName, Type queryType, String queryParameters, C
             case UNFINISHED:
                 return new UnfinishedJobsQuery(tableName);
             case DETAILED:
+                if (queryParameters == null) {
+                    throw new IllegalArgumentException("No parameters provided");
+                }
                 return DetailedJobsQuery.fromParameters(queryParameters);
             case RANGE:
                 return RangeJobsQuery.fromParameters(tableName, queryParameters, clock);
```

Comment:
It's not quite clear why queryParameters is not needed for the range query. I think I prefer having it modelled on the enum, so that we have the name to give you a clue that it's required for a detailed query but not for a range query. There might be a better way.

I'm also thinking about the case where we want to add another query type and the parameters are required. With the method on the enum, I'd see that and realise I need to add it there. With the check here I might not find that.